apps/src/music/player/MusicLibrary.ts

```diff
@@ -30,6 +32,37 @@ export default class MusicLibrary {
   getFolderForPath(path: string): SoundFolder | null {
     return this.groups[0].folders.find(folder => folder.path === path) || null;
   }
+
+  // A progression step might specify a smaller set of allowed sounds.
+  setAllowedSounds(allowedSounds: Sounds): void {
+    this.allowedSounds = allowedSounds;
+  }
+
+  // A sound picker might want to show the subset of sounds permitted by the
+  // progression's currently allowed sounds.
+  getAllowedSounds(folderType: string): Sounds {
+    const folders = this.groups[0].folders;
+
+    // Let's just do a deep copy and then do filtering in-place.
+    let foldersCopy = JSON.parse(JSON.stringify(folders));
+
+    // Whether or not we have allowedSounds, we need to filter by type.
+    foldersCopy = foldersCopy.filter(
+      (folder: any) => folder.type === folderType
+    );
```

Comment:
if you do want to retain the typing here, you can manually assign the parsed JSON the desired type, like `let foldersCopy: SoundFolder[] = JSON.parse(JSON.stringify(folders)) as SoundFolder[]` since we know this will be valid

And then in the `filter` function, you can type `folder` as `SoundFolder`

Reply:
Oops, I forgot to lint for `any` :).  Thanks for catching this.